src/3rd_party_add.c

```diff
@@ -0,0 +1,59 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#define _GNU_SOURCE
+#include "3rd_party_internal.h"
+#include "config.h"
+#include "lib/log.h"
+#include "swupd.h"
+#include "sys/stat.h"
+
+#ifdef THIRDPARTY
+
+static int create_repo_dir(char *repo_path)
+{
+	//repo_path doesn't exist
+	if (mkdir_p(repo_path) != 0 || chmod(repo_path, S_IRWXU) != 0) {
+		error("failed to create %s\n", repo_path);
+		return -1;
+	}
+
+	return SWUPD_OK;
+}
+
+int third_party_add(char *repo_path)
+{
+	if (repo_exists(repo_path)) {
```

Comment:
We need to have a file in /var/lib/swupd to track the 3rd-party repositories and the corresponding url, so checking for the dir is not enough. And if the dir is completly removed we should be able to restore that.

Reply:
Will address this in a follow-on PR as discussed